EIPS/EIP-1776.md

```diff
@@ -0,0 +1,187 @@
+
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL
+NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and
+"OPTIONAL" in this document are to be interpreted as described in
+RFC 2119.
+
+**ERC-721 compliant contracts MAY implement this ERC for royalties to provide a standard method of accepting royalty payments and receiving royalty information**
+
+The `_RoyaltyAmount` **MUST** be calculated as a percentage - such as "5" - for 5%. This is **REQUIRED** to maintain uniformity across the standard. 
+
+Marketplaces that implement the `Royalty` standard **MAY** implement any method of calculation or transfer of funds depending on their marketplace style and smart contract code. 
+
+```solidity
+pragma solidity ^0.6.0;
+import "./ERC165.sol";
+
+/**
+ * @dev Implementation of royalties for 721s
+ *
+ */
+abstract contract Royalties is ERC165 {
+    /*
+     * ERC165 bytes to add to interface array - set in parent contract implementing this standard
+     *
+     * bytes4(keccak256('royaltyInfo()')) == 0x46e80720
+     * bytes4 private constant _INTERFACE_ID_ERC721ROYALTIES = 0x46e80720;
+     * _registerInterface(_INTERFACE_ID_ERC721ROYALTIES);
+     */
+
+    uint256 private royalty_amount;
+    address private creator;
+    bytes4 private constant _INTERFACE_ID_ERC721ROYALTIES = 0x46e80720;
+    /**
+    @notice This event is emitted when royalties are transferred.
+
+    @dev The marketplace would emit this event from their contracts. Or they would call royaltiesReceived() function.
+
+    @param creator The original creator of the NFT entitled to the royalties
+    @param buyer The person buying the NFT on a secondary sale
+    @param amount The amount being paid to the creator
+    */
+    event ReceivedRoyalties(
+        address indexed creator,
+        address indexed buyer,
+        uint256 indexed amount
+    );
+
+    /**
+     *  @notice Constructor called from the NFT being deployed with the value for the royalty in percentage and the creator who will receive the royalty payment
+     *
+     *  @param _amount The percentage value on each sale that will be transferred to the creator
+     *  @param _creator The original creator of the NFT entitled to the royalties
+     *
+     */
+
+    constructor(uint256 _amount, address _creator) internal {
+        royalty_amount = _amount;
+        creator = _creator;
+        _registerInterface(_INTERFACE_ID_ERC721ROYALTIES);
+    }
+
+    /**
+     *      @notice Called to return the royalty amount that was set and only return that amount
+     */
+    function royaltyAmount() public view returns (uint256) {
+        return royalty_amount;
+    }
```

Comment:
No - I considered weighing the royalty per NFT - and there is a few pros and a few cons. 

Overall the main reason to go towards a royalty per contract (all the NFTs in this one contract) was due to gas costs during minting. 

Setting a royalty value per NFT works - but prohibits the ability to mint multiple NFTs in a single transaction - by increasing the state per NFT. I've redone the 721 standard for batch minting and can achieve upwards of 2,500 NFTs in a single transaction, but if royalties are set per NFT, it would be less than 40 per NFT. Same goes for setting a value like URI instead of using a chronological URI pattern and baseURI. 

I see the value prop of having a royalty per NFT, but there are downsides for the creator. If the creator wanted to set royalties for all their NFTs - in the long run of minting each NFT they would be paying more than if they were able to set the base royalty amount. 

Thinking about it further - you could handle it the same way as a baseURI and (this is purely implementation) return the base royalty amount for all NFTs, or individual amount if its set. 

Therefore the function could take a tokenId argument, and the returned value only needs to be a uint256, regardless if its a baseURI or not. 

Okay - I've convinced myself lol. Will adjust changed and allow for tokenId input. Makes sense as it offers more flexibility, albeit costing more gas during implementations.

Reply:
I think whether royalties are contract wide or not is an implementation detail.  One could imagine a token contract where all tokens have the same royalty, in which case no matter what tokenID you pass in you get the same result, but updating it merely requires updating a single variable.  Another token may have a default fee that can be overridden by individual tokens, and a third may have a fee rate per token.